Gemfile

```diff
@@ -147,6 +147,7 @@ def override_gem(name, *args)
     dependencies.delete(dependency)
 
     gem name, *args
+    Bundler::UI::Shell.new.warn "Caution, using override_gem for #{name} with args: #{args.inspect}, review Gemfile.dev.rb if this is not desired!"
```

Comment:
- I think it would be good to move it up one line.
- We probably don't want to see this in production and files in `bunder.d/`use it.
- I don't like the static reference to `Gemfile.dev.rb` since it could be `Gemfile.dev.rb` or any other file in `bundler.d/`, so it should reference the correct file if you want to keep the reference in there. (I think it's helpful)